### PR TITLE
Reinstate `region` to `default_args` for Alibaba example DAGs

### PR DESCRIPTION
**Pull request by josh-fell**
Related: #19891

In a previous PR to fix Mypy errors, the `region` arg was removed from `default_args` in the Alibaba example DAGs. While this did alleviate Mypy errors it removed a valid DAG-authoring feature that was being showcased.  This PR returns the `region` arg to `default_args` as well as handling Mypy errors.

> Note: Generally it seems like the first approach to handle these call-arg errors from Mypy related to `default_args` functionality is to add the ignore comments where appropriate. A proper fix for these scenarios seems very non-trivial and this is a good interim solution.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Should we make region Optional and raise Runtime Error instead as I did in https://github.com/apache/airflow/pull/20422 ?

**Comment by potiuk** [comment]
See: https://github.com/apache/airflow/pull/20422/files#diff-aec3e30af161525a7290fd9f420169f8a107c929e9b33b613bef4c03dd228c2dR70

**Comment by potiuk** [comment]
Ah nice. Using stub files seems to be a better solution. I will use that in #20422 too

**Comment by josh-fell** [comment]
Hmm. It seems that the Mypy checks in CI and running `mypy` locally in Breeze have different results. Still showing a call-arg error for `region` even with the stubs.

**Comment by potiuk** [comment]
> Hmm. It seems that the Mypy checks in CI and running `mypy` locally in Breeze have different results. Still showing a call-arg error for `region` even with the stubs.

Not for me #20422 It worked just fine in `Breeze` - same way as in CI. 

It could be about .pyc files probably ? (wild guess)  Did you actually use breeze or `mypy` in local virtualenv? And maybe you had some locally generated .pyc files ? Because in Breeze. `PYTHONDONTWRITEBYTECODE=true` is set, so .pyc files are not generated, but if you run it locally they coudl be

File: airflow/providers/alibaba/cloud/example_dags/example_oss_bucket.pyi [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from typing import Optional
+
+class OSSCreateBucketOperator:
+    def __init__(
+        self,
+        region: str = ...,
```
Comment (potiuk) [inline airflow/providers/alibaba/cloud/example_dags/example_oss_bucket.pyi]: Ah noe 

```suggestion
        region: Optional[str] = None
```

**Comment by potiuk** [comment]
> Hmm. It seems that the Mypy checks in CI and running `mypy --namespace-packages airflow/providers/alibaba/cloud` locally in a Breeze container have different results. Still showing a call-arg error for `region` even with the stubs.

One more comment - what worked for me, I set the fields to be "Optional[str] = None" in stubs. That was what worked perfectly for me.

File: airflow/providers/alibaba/cloud/example_dags/example_oss_bucket.pyi (hunk 1 above)
Comment (potiuk) [reply airflow/providers/alibaba/cloud/example_dags/example_oss_bucket.pyi]: And others.

Comment (josh-fell) [reply airflow/providers/alibaba/cloud/example_dags/example_oss_bucket.pyi]: Oh OK. Let me try this!

**Review by potiuk — APPROVED** [review]

**Comment by josh-fell** [comment]
There was a typo in an import within the stub file which was causing the stub to not be recognized. @potiuk can you review again when you get a chance please?

**Review by potiuk — APPROVED** [review]

**Pull request merged**